pkg/providers/instancetype/types.go

```diff
@@ -250,12 +250,13 @@ func habanaGaudis(info *ec2.InstanceTypeInfo) *resource.Quantity {
 // The number of pods per node is calculated using the formula:
 // max number of ENIs * (IPv4 Addresses per ENI -1) + 2
 // https://github.com/awslabs/amazon-eks-ami/blob/master/files/eni-max-pods.txt#L20
-func ENILimitedPods(info *ec2.InstanceTypeInfo) *resource.Quantity {
+func ENILimitedPods(ctx context.Context, info *ec2.InstanceTypeInfo) *resource.Quantity {
 	// VPC CNI only uses the default network interface
 	// https://github.com/aws/amazon-vpc-cni-k8s/blob/3294231c0dce52cfe473bf6c62f47956a3b333b6/scripts/gen_vpc_ip_limits.go#L162
 	networkInterfaces := *info.NetworkInfo.NetworkCards[*info.NetworkInfo.DefaultNetworkCardIndex].MaximumNetworkInterfaces
+	usableNetworkInterfaces := lo.Max([]int64{(networkInterfaces - int64(awssettings.FromContext(ctx).ReservedENIs)), 0})
```

Comment:
sanity check: in the case that this value is 0, I'm assuming that we error out during scheduling since none of the instance types would have sufficient resources to schedule pods.

Reply:
Yes, the max-pods count would actually be 2 if usableNetworkInterfaces is 0 which is correct for kube-proxy and vpc-cni. But we wouldn't launch a node that only has room for DaemonSets. I guess there are situations where kube-proxy is a static pod in which case maybe I'll just check if it's two and set it to 0 explicitly.

Reply:
updated to return 0 pods if usableNetworkInterfaces is 0.

Reply:
Can we add a comment around why we are explicitly setting it to 0 here?